client/my-sites/stats/activity-log-tasklist/to-update.jsx

```diff
@@ -44,11 +52,15 @@ export default WrappedComponent => {
 					{ ...this.props }
 					siteId={ this.props.siteId }
 					plugins={ this.state.plugins }
+					themes={ this.props.themes }
 				/>
 			);
 		}
 	}
-	return connect( ( state, { siteId } ) => ( {
-		plugins: getPluginsWithUpdates( state, [ siteId ] ),
-	} ) )( PluginsToUpdate );
+	return connect( ( state, { siteId } ) => {
+		return {
+			plugins: getPluginsWithUpdates( state, [ siteId ] ).map( mapUpdateNewVersionToVersion ),
+			themes: get( requestSiteAlerts( siteId ), 'data.updates.themes', [] ),
```

Comment:
you won't find may examples of this around calypso but I would recommend making some variable to return instead of `[]`. it is maybe more common in selectors…

```js
const emptyList = [];

…

themes: get( …, 'data.updates.themes', emptyList ),
```

the reason is that `[]` is created as a new value on every call to `connect()`, which is every render, so regardless if anything changes we force a re-render.

the same is true for `getPluginsWithUpdates( state, [ siteId ] ).map( … )` - that `map` creates new values on every render and will trigger needless re-renders. 

what's the purpose of `mapUpdateNewVersionToVersion` anyway? if we need more attributes on the data then I think it would be right to move that extension somewhere else, typically in the selector (and we have ways of caching these selectors so that they don't create new objects on every call too).

further, the `extend()` in there mutates the original object which we _really_ want to be careful of doing here because that can end up mutating our state tree which is a major issue since it can cause odd havoc in unexpected places around the app. for extending objects we like to use `{ ...source, newProp: value }` or `Object.assign( { newProp: value }, source )` and the like.

Reply:
Thanks for the pointer about the []. I've updated it in a few places.

The thing with mapUpdateNewVersionToVersion is that we don't need more attributes, it's just a way to avoid always checking that if the element is a plugin or a theme since plugin data has its version in `PLUGINNAME.update.new_version` and themes in `THEMENAME.version`.

I'm aware of the mutation and ok with it. We always need the version.